apps/astarte_appengine_api/lib/astarte_appengine_api/device/device.ex

```diff
@@ -775,6 +775,29 @@ defmodule Astarte.AppEngine.API.Device do
     end
   end
 
+  defp cast_value(:longinteger, string_value) when is_binary(string_value) do
+    with {int_value, _} <- Integer.parse(string_value) do
+      {:ok, int_value}
+    else
+      :error ->
+        {:error, :unexpected_value_type, expected: :longinteger}
+    end
```

Comment:
Since you have a single clause and you have to explicitly handle the `else` clause anyway, use a `case`:
```elixir
case Integer.parse(string_value) do
  # ...
end
```

Also note that you should match on `{int_value, ""}`, otherwise a value like `"12345foobar"` would pass the check (since `Integer.parse` returns `{12345, "foobar"}`)